### PR TITLE
filetype: some tools config files are not recognized

### PR DESCRIPTION
**Pull request by Freed-Wu**
Problem: Some tools config files are not recognized.
Solution: Add patterns for some tools config files
Refer:
- https://packaging.python.org/en/latest/specifications/pypirc/
- https://jorisroovers.com/gitlint/latest/configuration/
- https://pylint.pycqa.org/en/latest/user_guide/usage/run.html#command-line-options
- https://docs.bpython-interpreter.org/en/latest/configuration.html
- https://mypy.readthedocs.io/en/stable/config_file.html#the-mypy-configuration-file
- https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file

### CODE REVIEW

File: runtime/filetype.vim [hunk 1]
```diff
@@ -523,9 +523,16 @@ au BufNewFile,BufRead *.lrc			setf lyrics
 au BufNewFile,BufRead *.quake,cm3.cfg		setf m3quake
 au BufNewFile,BufRead m3makefile,m3overrides	setf m3build
 
-" Many Python tools use dosini as their config, like setuptools, pudb, coverage
+" Many tools written in Python use dosini as their config, like setuptools, pudb, coverage, pypi, gitlint, oelint-adv, pylint, bpython, mypy
 " (must be before *.cfg)
-au BufNewFile,BufRead setup.cfg,pudb.cfg,.coveragerc	setf dosini
+au BufNewFile,BufRead setup.cfg,pudb.cfg,.coveragerc,.pypirc,.gitlint,.oelint.cfg,{.,}pylintrc,*/bpython/config,*/mypy/config	setf dosini
```
Comment (chrisbra) [inline runtime/filetype.vim]: can you split this up please?

File: src/testdir/test_filetype.vim [hunk 2]
```diff
@@ -210,7 +210,7 @@ def s:GetFilenameChecks(): dict<list<string>>
     dnsmasq: ['/etc/dnsmasq.conf', '/etc/dnsmasq.d/file', 'any/etc/dnsmasq.conf', 'any/etc/dnsmasq.d/file'],
     dockerfile: ['Containerfile', 'Dockerfile', 'dockerfile', 'file.Dockerfile', 'file.dockerfile', 'Dockerfile.debian', 'Containerfile.something'],
     dosbatch: ['file.bat'],
-    dosini: ['/etc/yum.conf', 'file.ini', 'npmrc', '.npmrc', 'php.ini', 'php.ini-5', 'php.ini-file', '/etc/yum.repos.d/file', 'any/etc/yum.conf', 'any/etc/yum.repos.d/file', 'file.wrap', 'file.vbp', 'ja2.ini', 'JA2.INI', 'setup.cfg', 'pudb.cfg', '.coveragerc'],
+    dosini: ['/etc/yum.conf', 'file.ini', 'npmrc', '.npmrc', 'php.ini', 'php.ini-5', 'php.ini-file', '/etc/yum.repos.d/file', 'any/etc/yum.conf', 'any/etc/yum.repos.d/file', 'file.wrap', 'file.vbp', 'ja2.ini', 'JA2.INI', 'setup.cfg', 'pudb.cfg', '.coveragerc', '.pypirc', '.gitlint', '.oelint.cfg', 'pylintrc', '.pylintrc', '/home/user/.config/bpython/config', '/home/user/.config/mypy/config'],
```
Comment (chrisbra) [inline src/testdir/test_filetype.vim]: Same here, please start splitting this up.

**Review by chrisbra — CHANGES_REQUESTED** [review]

File: runtime/filetype.vim (hunk 1 above)
Comment (Freed-Wu) [reply runtime/filetype.vim]: Done.

File: src/testdir/test_filetype.vim (hunk 2 above)
Comment (Freed-Wu) [reply src/testdir/test_filetype.vim]: Done.

File: src/testdir/test_filetype.vim [hunk 3]
```diff
@@ -212,7 +212,14 @@ def s:GetFilenameChecks(): dict<list<string>>
     dnsmasq: ['/etc/dnsmasq.conf', '/etc/dnsmasq.d/file', 'any/etc/dnsmasq.conf', 'any/etc/dnsmasq.d/file'],
     dockerfile: ['Containerfile', 'Dockerfile', 'dockerfile', 'file.Dockerfile', 'file.dockerfile', 'Dockerfile.debian', 'Containerfile.something'],
     dosbatch: ['file.bat'],
-    dosini: ['/etc/yum.conf', 'file.ini', 'npmrc', '.npmrc', 'php.ini', 'php.ini-5', 'php.ini-file', '/etc/yum.repos.d/file', 'any/etc/yum.conf', 'any/etc/yum.repos.d/file', 'file.wrap', 'file.vbp', 'ja2.ini', 'JA2.INI', 'mimeapps.list', 'pip.conf', 'setup.cfg', 'pudb.cfg', '.coveragerc', 'psprint.conf', 'sofficerc', 'any/.config/lxqt/globalkeyshortcuts.conf', 'any/.config/screengrab/screengrab.conf'],
+    dosini: ['/etc/yum.conf', '/etc/nfs.conf', '/etc/nfsmount.conf', 'file.ini',
+             'npmrc', '.npmrc', 'php.ini', 'php.ini-5', 'php.ini-file',
```
Comment (chrisbra) [inline src/testdir/test_filetype.vim:216]: missing backslash

**Review by chrisbra — CHANGES_REQUESTED** [review]

File: src/testdir/test_filetype.vim (hunk 3 above)
Comment (chrisbra) [reply src/testdir/test_filetype.vim:216]: ah sorry, it's vim9 script, so should be fine

**Comment by Freed-Wu** [comment]
Fixed.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**